### DISCUSSION TITLE
How to use plugins plugins with npx without installing

### DISCUSSION THREAD

**Discussion by mehdihadeli** [semantic-release/semantic-release#2674] [Q&A] [answered]
Hi,
Could we use npx plugins together with `npx semantic-release` without installing?
I don't have a JS project and I don't want to install node_modules.

**Comment by gr2m** [comment]
What I usually do is to install the plugin in the release workflow (e.g. `npm install my-semantic-release-plugin`, right before running `npx semantic-release`.

Ultimately we are thinking about creating a `@semantic-release/core` package which can easily be composed with plugins into custom semantic-release CLIs, so you'd be able to simply run `npx my-semantic-release` and it will have all the plugins and configurations you want compiled into an npm package. But no work has been done towards that, we want to conclude the ESM transition first

**Comment by mehdihadeli** [comment, marked as answer]
Thanks for your response,
I did something like this and it works.

``` bash
npx -p @semantic-release/git -p @semantic-release/changelog -p @semantic-release/exec  semantic-release 
```

**Reply by gr2m** [reply]
Nice! I didn't know you can do that 👍🏼

**Reply by arcticShadow** [reply]
This is going to change my world thanks!
